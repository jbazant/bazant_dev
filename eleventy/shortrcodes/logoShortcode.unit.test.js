const { logoShortcode } = require('./logoShortcode');
describe('logoShortcode', function () {
  it('should match snapshot', function () {
    expect(logoShortcode(100, 'white', 'black')).toMatchSnapshot();
  });

  it('should have correct size and scale', function () {
    const logo = logoShortcode(100, 'white', 'black');

    expect(logo).toMatch('width="100"');
    expect(logo).toMatch('height="100"');
    expect(logo).toMatch('transform="scale(0.5)"');
  });

  it('should have correct color', function () {
    const logo = logoShortcode(100, 'blue', 'purple');

    expect(logo).toMatch('fill="blue"');
    expect(logo).toMatch('stroke="purple"');
  });
});
