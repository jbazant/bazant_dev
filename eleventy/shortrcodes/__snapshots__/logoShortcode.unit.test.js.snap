// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logoShortcode should match snapshot 1`] = `
"<svg 
    version="1.1"
    baseProfile="full"
    width="100" height="100"
    xmlns="http://www.w3.org/2000/svg">
      <g transform="scale(0.5)">
        <path
          d="m 24.5 143 l 150 0 l -75 -129.9038 z m 19.0021 -32.4135 l 37.1398 0 l -18.5699 32.164 z m 55.9979 -32.6632 l -18.5699 -32.164 l 37.1398 0 z m 0 64.8272 l -37.2839 -64.5775 l 74.5678 -0.0001 z m 37.428 0 l -18.5699 -32.164 l 37.1398 0 z"
          fill="white"
          stroke="none"
        />
        <circle 
          r="90"
          stroke="black"
          fill="none"
          stroke-width="20"
          stroke-dasharray="392 20 26 20 26 20 26 20"
          transform="translate(100, 100) rotate(30)"
        />
      </g>
    </svg>"
`;
